# V.Known sites preparation for BQSR 

This document describes the preparation of known variant sites and the execution of *Base Quality Score Recalibration (BQSR)* using *GATK*for *human chromosome 7 (hg38)*.

BQSR corrects systematic errors in base quality scores generated by sequencing machines, improving the accuracy of downstream variant calling.

---

## V.1 - Known Sites Preparation for BQSR

 - BQSR requires a set of *known, high-confidence variants* to identify and model sequencing errors.
 - For human whole-genome analysis, *Mills and 1000G gold standard indels (hg38)* are commonly used.

## V.2 - Create Known Sites Directory

 - Creates a dedicated directory to store known variant VCF files required for BQSR.

```bash
mkdir -p Known_Sites
```
## V.3 - Download Known Variant Sites (hg38)

 - Downloads the *Mills and 1000G indels VCF file* and corresponding *Tabix index (.tbi) file*
 - These files represent high-confidence indel variants used as known sites

**Commmand**
```bash
wget https://storage.googleapis.com/genomics-public-data/resources/broad/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz
wget https://storage.googleapis.com/genomics-public-data/resources/broad/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi
```
 - This command generates output files of *Mills_and_1000G_gold_standard.indels.hg38.vcf.gz, Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi*
   
## V.4 - Move Files to Known_Sites Directory

**Command**
```bash
mv Mills_and_1000G_gold_standard.indels.hg38.* Known_Sites/
```
## V.5 - Restrict Known Sites to Chromosome 7

 - Since this project focuses only on chromosome 7, the known-sites VCF is restricted to chr7

**Command**
```bash
bcftools view -r 7 \
  Known_Sites/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz \ -Oz -o Known_Sites/Mills_chr7.vcf.gz
```
 - This command generates the file of Mills_chr7.vcf.gz
   
## V.6 - Index the Chromosome 7 Known Sites VCF

 - A Tabix-compatible index is generated using GATK.

**Command**
```bash
sudo docker run \
  -v $(pwd):/data \ broadinstitute/gatk:latest gatk IndexFeatureFile \ -I /data/Known_Sites/Mills_chr7.vcf.gz
```
 - This command generates the file of *Mills_chr7.vcf.gz.csi* (Coordinate Sorted Index file)
   
## V.7 - Convert .csi Index to .tbi (Standardization)

 - GATK generated a .csi index for the chromosome-restricted VCF.
 - Although .csi is fully supported by GATK, the index was converted to .tbi for standardization and compatibility with other tools.

**Command**
```bash
tabix -f -p vcf Known_Sites/Mills_chr7.vcf.gz
```
 - This command generates the file of *Mills_chr7.vcf.gz.tbi*

## V.8 - Files generated in this step

 - Output_Files/Known_Sites
    - Mills_and_1000G_gold_standard.indels.hg38.vcf.gz
    - Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi
    - Mills_chr7.vcf.gz
    - Mills_chr7.vcf.gz.csi
    - Mills_chr7.vcf.gz.tbi



